src/main/java/spinnery/widget/WStatusBar.java

```diff
@@ -0,0 +1,139 @@
+package spinnery.widget;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import spinnery.client.utility.SpriteSheet;
+import spinnery.widget.api.*;
+
+/**
+ * Used to render a status bar, such as a health or hunger bar. Visuals are specified using a {@link BarConfig}.
+ */
+@Environment(EnvType.CLIENT)
+public class WStatusBar extends WAbstractWidget {
+    private static final SpriteSheet ICONS = SpriteSheet.MINECRAFT_ICONS;
+    /**
+     * The normal health bar without any status modifiers.
+     */
+    public static final BarConfig HEALTH_DEFAULT = new BarConfig(ICONS.getSprite(0), ICONS.getSprite(4), ICONS.getSprite(5), BarConfig.Direction.LeftToRight);
+    public static final BarConfig HEALTH_WITHER = new BarConfig(ICONS.getSprite(0), ICONS.getSprite(12), ICONS.getSprite(13), BarConfig.Direction.LeftToRight);
+    public static final BarConfig HEALTH_POISON = new BarConfig(ICONS.getSprite(0), ICONS.getSprite(8), ICONS.getSprite(9), BarConfig.Direction.LeftToRight);
+    /**
+     * The normal food bar. Note that this is a right to left bar by default. You can mirror it using {@link BarConfig#mirror()}.
+     */
+    public static final BarConfig FOOD_DEFAULT = new BarConfig(ICONS.getSprite(54), ICONS.getSprite(58), ICONS.getSprite(59), BarConfig.Direction.RightToLeft);
+    /**
+     * Food bar with the hunger status effect.
+     */
+    public static final BarConfig FOOD_HUNGER = new BarConfig(ICONS.getSprite(56), ICONS.getSprite(62), ICONS.getSprite(63), BarConfig.Direction.RightToLeft);
+    public static final BarConfig ARMOR_DEFAULT = new BarConfig(ICONS.getSprite(18), ICONS.getSprite(20), ICONS.getSprite(19), BarConfig.Direction.LeftToRight);
+
+    public static class BarConfig {
+        // Always renders, even behind full sprites.
+        public final SpriteSheet.Sprite background;
+        public final SpriteSheet.Sprite full;
+        public final SpriteSheet.Sprite half;
+        // Some bars are specified in their atlas using a right-to-left texture, such as the hunger bar icons.
+        public final Direction direction;
+        // This allows you to flip a right to left bar around to make it left to right.
+        public final boolean mirror;
+        // These are 9 for all Minecraft bars, but you can change them if you're making a custom bar.
+        public final int spriteWidth;
+        public final int spriteHeight;
+
+        public enum Direction {
+            LeftToRight,
```

Comment:
Swap for `LEFT_TO_RIGHT` & `RIGHT_TO_LEFT` to keep consistent with the internal style.